code-challenge-android-kotlin/app/src/main/java/com/arctouch/codechallenge/home/HomeAdapter.kt

```diff
@@ -36,5 +44,28 @@ class HomeAdapter(private val movies: List<Movie>) : RecyclerView.Adapter<HomeAd
 
     override fun getItemCount() = movies.size
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) = holder.bind(movies[position])
+    override fun onBindViewHolder(holder: ViewHolder, position: Int){
+        holder.bind(movies[position])
+        holder.movieRelativeLayout.setOnClickListener {
+            showProgressbar()
+            openMovieFragment(movies[position].id)
+        }
+    }
+
+    fun openMovieFragment(id: Int){
+        //TODO remove API call
+        val fragment = MovieFragment.newInstance(id, closeMovieFragment)
+        val fragmentTransaction = fragmentManager.beginTransaction()
+        if(firstFragment == false) fragmentTransaction.add(R.id.frameLayout, fragment,"MOVIE_FRAGMENT")
```

Comment:
Here you're using the `"MOVIE_FRAGMENT"` string twice. It should be better to declare a companion object and a const enclosed by the MovieFragment class, like following:

    public class MovieFragment : Fragment {
        companion object {
            const val TAG = "MovieFragment.TAG"
        }

And use it like that:

    if(firstFragment == false) fragmentTransaction.add(R.id.frameLayout, fragment, MovieFragment.TAG) else fragmentTransaction.replace(R.id.frameLayout, fragment, MovieFragment.TAG)

Reply:
Thanks for the advice. I had no idea where to declare this kind of constant. :)